### ISSUE TITLE
[QUERY] How to create Storage Account with specific properties set ( ex :  allowCrossTenantReplication  )

### ISSUE THREAD

**Issue by x202014** [Azure/azure-sdk-for-java#37436]
**Query/Question**
A clear and concise ask/query.

Hello,
 we have to create storage accounts with java sdk.
 We are using the following sample code : 

```java
resourceManager.storageAccounts()
	.define("testaccount")
	.withRegion("westeurope")
	.withExistingResourceGroup("myresourceGroup")
	.withAccessFromSelectedNetworks()
	.withSku(StorageAccountSkuType.fromSkuName("STANDARD_LRS")
	.withOnlyHttpsTraffic()
	.withMinimumTlsVersion(MinimumTlsVersion.fromString("TLS1_2"))
	.withHnsEnabled(true)
	.withGeneralPurposeAccountKindV2()
	.create()
```

We want to set the property allowCrossTenantReplication  to false.

the field seems available:  StorageAccountPropertiesCreateParameters.

 We successed to set it with reflection ( but we don't want to set it this way ): 
```java
WithCreate c = azureResourceManager.storageAccounts().define(saName)
       .withRegion(workspace.getRegion())
       .withExistingResourceGroup(workspace.getResourceGroup())

       .withMinimumTlsVersion(MinimumTlsVersion.TLS1_2)
       .withTags(getTags())
       .withOnlyHttpsTraffic()
       .withGeneralPurposeAccountKindV2()
       .withSku(StorageAccountSkuType.STANDARD_RAGRS)
       .withHnsEnabled(true);

	// Workaround to force allow cross tenant replication to false
	Field f = c.getClass().getDeclaredField("createParameters");
	f.setAccessible(true);
	StorageAccountCreateParameters p = (StorageAccountCreateParameters) f.get(c);
	p.withAllowCrossTenantReplication(false);

   c.create();
```

We also find a way to set the security property : "Default to Azure Active Directory authorization in the Azure portal"

Thank you for your Help.
Nicolas.

***Why is this not a Bug or a feature Request?***
The possibility to set the field is available with .Net/Powershell , ARM template  according to documention ([powershell](https://learn.microsoft.com/en-us/powershell/module/az.storage/new-azstorageaccount?view=azps-10.4.1))

**Setup (please complete the following information if applicable):**
 - OS: Windows
 - IDE:  IntelliJ IDEA 2023.2.4 (Ultimate Edition)
 - Library/Libraries: azure-resourcemanager-storage-2.32.0.jar
 -  
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Query Added
- [x] Setup information Added

**Comment by alzimmermsft** [comment]
Thank you for filing this @x202014.

@weidongxu-microsoft could you follow-up on this?

**Comment by weidongxu-microsoft** [comment]
@XiaofeiCao Could we add it in Nov. release? You may see if Hong can do it.

**Comment by XiaofeiCao** [comment]
@x202014 "Default to Azure Active Directory authorization in the Azure portal" should be `defaultToOAuthAuthentication`:
![image](https://github.com/Azure/azure-sdk-for-java/assets/92354331/ced1c37b-19ba-482e-b18c-7c722b25b81b)
![image](https://github.com/Azure/azure-sdk-for-java/assets/92354331/5fafedb6-df9b-4ba8-bf7e-5f015eb3f7da)
![image](https://github.com/Azure/azure-sdk-for-java/assets/92354331/5b425eb0-9b81-47e5-bb0d-7765848921d1)

**Comment by x202014** [comment]
@XiaofeiCao , thank you for the information.
from the portal , in powershell, we are able to set those properties. 
My concert is regarding Java, we can't access them easily.
Thanks all for your help.
Nicolas.

**Comment by x202014** [comment]
Nice, noted, thank you for your reactivity.

**Comment by XiaofeiCao** [comment]
@x202014 We've released 2.33.0 in Nov. Sorry for the late reply.

**Comment by x202014** [comment]
Hello,
thank you for the feedback.
Nicolas.
